modules/storages/app/contracts/storages/project_storages/base_contract.rb

```diff
@@ -0,0 +1,51 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2022 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+# A "contract" is an OpenProject patter used to validate parameters
+# before actually creating a model.
+# Used by: projects_storages_controller.rb and in the API
+module Storages::ProjectStorages
+  class BaseContract < ::ModelContract
```

Comment:
I just checked, and `::ModelContract` is inheriting `::BaseContract`, so it will ensure the other behaviors you mentioned. The added behavior is that it validates the model is valid and merges model errors into its own contract errors.

Are you sure we should change the parent class from `::ModelContract` to `::BaseContract` here?